src/ReachSets/ContinuousPost/BFFPS19/reach_blocks.jl

```diff
@@ -0,0 +1,270 @@
+                       δ::NUM,
+                       guards_proj::Vector{<:LazySet{NUM}},
+                       block_options,
+                       vars::Vector{Int},
+                       termination::Function,
+                       progress_meter::Union{Progress, Nothing},
+                       res::Vector{<:ReachSet}
+                       )::Tuple{Int, Bool} where {NUM}
+    update!(progress_meter, 1)
+    array = CartesianProductArray(Xhat0[blocks])
+    X_store = (output_function == nothing) ?
+        array :
+        box_approximation(output_function(array))
+    t0 = zero(δ)
+    t1 = δ
+
+    diff_blocks = setdiff(1:length(partition),blocks)
+
+    b = length(blocks)
+    bd = length(diff_blocks)
+
+    array_d = CartesianProductArray(Xhat0[diff_blocks])
+    X_store_d = (output_function == nothing) ?
+            array_d :
+            box_approximation(output_function(array_d))
+    X_store = combine_cpas(X_store, X_store_d, blocks, diff_blocks)
+    store!(res, 1, X_store, t0, t1, N)
+    terminate, skip = termination(1, X_store, t0)
+    if terminate
+        return 1, skip
+    end
+
+    b = length(blocks)
+    Xhatk = Vector{LazySet{NUM}}(undef, b)
+    Xhatk_d = Vector{LazySet{NUM}}(undef, bd)
+    ϕpowerk = copy(ϕ)
+
+    if U != nothing
+        Whatk_blocks = Vector{LazySet{NUM}}(undef, b)
+        Whatk_diff_blocks = Vector{LazySet{NUM}}(undef, bd)
+        inputs = next_set(U)
+        @inbounds for i in 1:b
+            bi = partition[blocks[i]]
+            Whatk_blocks[i] = overapproximate_inputs(1, blocks[i], proj(bi, n) * inputs)
+        end
+        @inbounds for i in 1:bd
+            bi = partition[diff_blocks[i]]
+            Whatk_diff_blocks[i] = overapproximate_inputs(1, diff_blocks[i], proj(bi, n) * inputs)
+        end
+    end
+
+    k = 2
+    @inbounds while true
+        update!(progress_meter, k)
+        X_store = deco_post_sparse(b, blocks, Whatk_blocks, partition,
+                                          ϕpowerk, Xhat0, output_function)
+
+        t0 = t1
+        t1 += δ
+
+        terminate, skip, reach_set_intersected = termination(k, X_store, t0)
+
+        if !(isdisjoint(X_store, UnionSetArray(guards_proj)))
+            X_store_d = deco_post_sparse(bd, diff_blocks, Whatk_diff_blocks, partition,
+                                              ϕpowerk, Xhat0, output_function)
+
+            rs_oa = Approximations.overapproximate(reach_set_intersected, CartesianProductArray, block_options)
+            X_store = combine_cpas(rs_oa, X_store_d, blocks, diff_blocks)
+        end
+
+        store!(res, k, X_store, t0, t1, N)
+
+        if U != nothing
+            for i in 1:b
+                bi = partition[blocks[i]]
+                Whatk_blocks[i] = overapproximate_inputs(k, blocks[i],
+                    Whatk_blocks[i] + row(ϕpowerk, bi) * inputs)
+            end
+        end
+
+         ##for diffs
+        if U != nothing
```

Comment:
Again you can merge this branch with the previous one (and also in the dense method).